### PR TITLE
[2.11.x]DDF-3440 Upgrades to Solr 6.6.2

### PR DESCRIPTION
**Pull request by vinamartin**
#### What does this PR do?

- Upgrades to Solr 6.6.2
- Removes Solr 6.6.1 owasp suppressions 
- Reverts change that disabled Solr Config API (https://github.com/codice/ddf/pull/2509/files)

Solr 6.6.2 Release Notes: https://lucene.apache.org/solr/6_6_2/changes/Changes.html#v6.6.2
Previous Solr Upgrade 6.6.1: https://github.com/codice/ddf/pull/2305

I've manually verified there have been no webapp changes or dependencies that need to be changed.

#### Who is reviewing it? 
@mcalcote @ahoffer @emanns95 @clockard @millerw8 @stustison 
@coyotesqrl @brjeter 

#### Choose 2 committers to review/merge the PR. 
@figliold
@lessarderic

#### How should this be tested?
(Instructions from the last upgrade)
Full build.
Configure it to use stand-alone solr, embedded solr, and solr cloud. In all three scenarios, confirm ability to ingest metacards, save workspaces, and query.

#### What are the relevant tickets?
[DDF-3440](https://codice.atlassian.net/browse/DDF-3440)

#### Notes on Review Process
For the general case, merging is gated on the following:

* Review by at least two contributors
* Review by at least two committers
* Successful CI
* Successful static analysis (SonarQube/Coverity/etc.)
* Heroing of change (details vary by ticket)

Certain exceptions may occur. For example, if there is a problem with the CI environment then a manually performed full build and static analysis check may be substituted _at the discretion of the repository maintainers_. Additionally, repository metadata such as this `PULL_REQUEST_TEMPLATE`, the repository `README`, and other non-functional items may be merged without requiring CI and static analysis.

Finally, an abbreviated review process may be in place for forward-port tickets immediately following branching and prior to branch divergence. In that case, approvals, CI, static analysis, and heroing that occur against the PR against the _release branch_ can be considered sufficient for reviewing the PR against the _master branch_.

Once the branches diverge, the process will return to normal.

Insofar as divergence is not a function of time but of change to the parent branch, the repository maintainers will determine when abbreviated review ends.

When abbreviated review is in effect, a note to that effect will be at the top of this template. Following is an example note:
```
##### ABBREVIATED REVIEW BETWEEN 2.12.X AND MASTER IS IN EFFECT
____

```

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by vinamartin** [comment]
build now

File: dependency-check-maven-config.xml [hunk 1]
```diff
@@ -183,13 +183,4 @@
         <cve>CVE-2016-2109</cve>
     </suppress>
 
-    <!-- Remove this suppression when DDF-3440 is worked -->
-    <suppress>
```
Comment (clockard) [inline dependency-check-maven-config.xml]: We will need to remove this suppression in downstream projects as well

**Comment by vinamartin** [comment]
```
19:43:04 [ERROR] Failed to execute goal org.owasp:dependency-check-maven:1.4.5:check (default) on project catalog-core-commons:
19:43:04 [ERROR] 
19:43:04 
[ERROR] One or more dependencies were identified with vulnerabilities that have a CVSS score greater then '7.0':
19:43:04 [ERROR] 
19:43:04 [ERROR] xercesImpl-2.11.0.jar: CVE-2012-0881
```
From the build results.

**Comment by brjeter** [comment]
I'm honestly not sure where that CVE came from. I had run OWASP last week and got no such CVE reported by OWASP. And that particular jar is definitely there when I run `mvn dependency:tree`
```
[INFO] |  \- xerces:xercesImpl:jar:2.11.0:compile
```
It's difficult to tell whether or not this applies to that version of xerces. Looks like the 2.11.0 version was released about a year after the CVE was reported.

**Comment by brjeter** [comment]
PR to resolve this issue: https://github.com/codice/ddf/pull/2703

**Comment by vinamartin** [comment]
build now

**Comment by vinamartin** [comment]
I wasn't able to reproduce the itest issues above locally, so for now I will rebuild. 
If they reoccur, I will investigate further.

build now

**Comment by ahoffer** [comment]
Built on Mac. Started it. Ingested and removed XML metacards from the CLI. Tried the Solr admin web interface, did some queries. The things I tried worked as expected.

**Comment by vinamartin** [comment]
build now

**Pull request merged**